src/stapi_fastapi/models/order.py

```diff
@@ -6,6 +6,19 @@
 
 from stapi_fastapi.models.shared import Link
 from stapi_fastapi.types.datetime_interval import DatetimeInterval
+from stapi_fastapi.types.filter import CQL2Filter
+
+
+class OrderParametersBase(BaseModel): ...
+
+
+class OrderRequest(BaseModel):
+    datetime: DatetimeInterval
+    geometry: Geometry
+    # TODO: validate the CQL2 filter?
+    filter: CQL2Filter | None = None
+    order_parameters: OrderParametersBase | None = None
+    model_config = ConfigDict(strict=True)
```

Comment:
This is just an `OpportunityRequest` with the addition of `order_parameters`? If so we should inherit from the former.

I don't think  order_parameters should be optional. I think I see the rationale (in the event there are no order parameters?), but this has the effect of making any required fields in OrderParametersBase be optional as well--because the user could just omit `order_parameters` entirely from their request. So we need to find another way to deal with this issue.

Also, the type here is wrong being `OrderParametersBase`. We need to make this a generic so we can pass in a subclass of `OrderParametersBase`. And this means we need to change the `POST /order` endpoint of `ProductRouter` to take with a fully-typed request model for the given product. FastAPI exposes a way to deal with this for responses via `response_model`, but doesn't provide such a knob for us to tweak for the request model. I think we can hack something together by overriding the function annotation in the `ProductRouter.__init__()`, but that might require some experimentation to get right.

Last thing: why strict? I'm not sure I see that being a valuable configuration? Maybe that's "inherited" from `OpportunityRequest`?

Reply:
>I don't think order_parameters should be optional. I think I see the rationale (in the event there are no order parameters?), but this has the effect of making any required fields in OrderParametersBase be optional as well--because the user could just omit order_parameters entirely from their request. So we need to find another way to deal with this issue.

Thinking more here, I think we could address this by doing `order_parameters: P = Field(default_factory=P)`, where `P` is the `TypeVar` for `OrderParametersBase` we've used to make `OrderRequest` a generic.